### ISSUE TITLE
error defining a new loss function?

### ISSUE THREAD

**Issue by cheng6076** [twitter/torch-autograd#100]
Hi, my goal is simply to define a new error function with autograd. I tested the following example

```lua
require 'nn'
grad = require 'autograd'
local mse = function(input, target) 
    local buffer = input-target
    return torch.sum( torch.cmul(buffer, buffer) ) / (input:dim() == 2 and input:size(1)*input:size(2) or input:size(1))
end
local autoMseCriterion = grad.nn.AutoCriterion('AutoMSE')(mse)
```

but got the following error: 

> while creating metatable autograd.nn.criterion.AutMSE: bad argument #1 (autograd is an invalid module name)

Can anyone help me? Thanks!

**Comment by alexbw** [comment]
Try requiring autograd first, and not requiring nn at all.

**Comment by cheng6076** [comment]
Thanks for the quick reply. I have tried both, but still got the same error, unfortunately.

**Comment by alexbw** [comment]
I will try to get to this issue by the end of this weke.

**Comment by alexbw** [comment]
```
grad = require 'autograd'
require 'nn'
local mse = function(input, target)
    local buffer = input-target
    return torch.sum( torch.cmul(buffer, buffer) ) / (input:dim() == 2 and input:size(1)*input:size(2) or input:size(1))
end
local autoMseCriterion = grad.nn.AutoCriterion('AutoMSE')(mse)
```

This code works with no issue on a fresh install of autograd. Please do a fresh install by using luarocks to first remove autograd, and then build again from a fresh pull from github.
